network/p2p/capabilities.go

```diff
@@ -121,6 +122,15 @@ func (c *CapabilitiesDiscovery) AdvertiseCapabilities(capabilities ...Capability
 		}()
 
 		for {
+			// shuffle capabilities to advertise in random order
+			// since the DHT's internal advertisement happens concurrently for peers in its routing table
+			// any peer error does not prevent advertisement of other peers.
```

Comment:
this would reduce the number of, rather than eliminate advertisement errors more likely?

Reply:
Shuffling would not reduce number of logged errors (the err -> warn and timeout will do) but will increase probability of more peers knowing of us exposing both gossip and archival capabilities.